hawk/app/controllers/tickets_controller.rb

```diff
@@ -132,51 +132,85 @@ def destroy
   end
 
   def show
+    @ticket = @cib.tickets[params[:id]]
+
     respond_to do |format|
       format.json do
         render json: @ticket.to_json
       end
       format.html
-      format.any { not_found  }
+      format.any { not_found }
     end
   end
 
   def grant
-    @ticket.grant! @cib.booth.me
-
-    respond_to do |format|
-      format.html do
-        flash[:success] = _("Successfully granted the ticket")
-        redirect_to cib_tickets_url(cib_id: @cib.id)
+    site = @cib.booth[:me]
+    if grant_ticket(params[:ticket], site)
+      respond_to do |format|
+        format.html do
+          flash[:success] = _("Successfully granted the ticket")
+          redirect_to cib_tickets_url(cib_id: @cib.id)
+        end
+        format.json do
+          render json: { success: true, message: _("Successfully granted the ticket") }
+        end
       end
-      format.json do
-        render json: {
-          success: true,
-          message: _("Successfully granted the ticket")
-        }
+    else
+      respond_to do |format|
+        format.html do
+          flash[:alert] = _("Error granting the ticket")
+          redirect_to cib_tickets_url(cib_id: @cib.id)
+        end
+        format.json do
+          render json: { error: _("Error granting the ticket") }, status: :unprocessable_entity
+        end
       end
     end
   end
 
   def revoke
-    @ticket.revoke! @cib.booth.me
-
-    respond_to do |format|
-      format.html do
-        flash[:success] = _("Successfully revoked the ticket")
-        redirect_to cib_tickets_url(cib_id: @cib.id)
+    if @ticket.revoke! params[:ticket]
+      respond_to do |format|
+        format.html do
+          flash[:success] = _("Successfully revoked the ticket")
+          redirect_to cib_tickets_url(cib_id: @cib.id)
+        end
+        format.json do
+          render json: {
+            success: true,
+            message: _("Successfully revoked the ticket")
+          }
+        end
       end
-      format.json do
-        render json: {
-          success: true,
-          message: _("Successfully revoked the ticket")
-        }
+    else
+      respond_to do |format|
+        format.html do
+          flash[:alert] = _("Error revoking the ticket")
+          redirect_to cib_tickets_url(cib_id: @cib.id)
+        end
+        format.json do
+          render json: { error: _("Error revoking the ticket") }, status: :unprocessable_entity
+        end
       end
     end
   end
 
   protected
 
+  def grant_ticket(ticket, site)
+    raise Constraint::CommandError.new _("Simulator active: Use the ticket controls in the simulator") if current_cib.sim?
+    out, err, rc = Invoker.instance.run "booth", "client", "grant", "-t", ticket, "-s", site.to_s
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `out`. Use `_` or `_out` as a variable name to indicate that it won't be used.